### DISCUSSION TITLE
Change content tab anchor links

### DISCUSSION THREAD

**Discussion by caseyamp** [squidfunk/mkdocs-material#4360] [Q&A] [answered]
Thanks for implementing [content tab anchor link feature](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#anchor-links)! 

Is there a way to manually set the id for these links? We'd like to be able to set them something a bit more meaningful than `www.example.com/#__tabbed_1_2`.

**Comment by squidfunk** [comment]
Unfortunately not. The reason is that [Tabbed](https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/) doesn't play together with the [Attribute Lists](https://python-markdown.github.io/extensions/attr_list/) extension. This was discussed before with @facelessuser, who maintains the Tabbed extension that generates the content tabs markup. __Related__: 

- #4268
- https://github.com/Python-Markdown/markdown/issues/1175

**Reply by caseyamp** [reply]
Ah thank you! I searched around for previous discussions but missed this.

**Reply by squidfunk** [reply]
@facelessuser wouldn't it make sense to allow for passing the content of a tab label through a slugify function? I'm working with [slugify functions](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.post_slugify) in the new [built-in blog plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/), and this might make support for Attribute Lists obsolete. Example:


``` markdown
=== "This is a label"

    Content
```

Might generate:

``` html
...
<input id="this-is-a-label" name="__tabbed_1" type="radio">
...
```

**Reply by facelessuser** [reply]
Well considering the direction that we may be taking with generalized blocks, attribute lists may not be needed in future work anyways.

As for sluggifying tab ids, I would need to look into to it closer. You can create an issue so that I can explore it. Whether this would make it into the existing solution or the new solution would have to be considered as well.

**Reply by squidfunk** [reply]
https://github.com/facelessuser/pymdown-extensions/issues/1807

**Comment by squidfunk** [comment, marked as answer]
__Update__: with the help of @facelessuser, content tabs can now be slugified which produces much more readable anchor links! Personally, I find it's a very good solution that involves no effort by the author. We've enabled it on our documentation. Our [documentation has been updated](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#anchor-links) and explains how to enable it. I'm also marking this as the new answer.

**Reply by facelessuser** [reply]
Very cool!

**Reply by caseyamp** [reply]
Bravo! Thank you so so so so so much for doing this.
